### PR TITLE
add fault to repository repository funcs

### PR DESCRIPTION
**Pull request by ryanfkeepers**
## Description

adds fault.Errors to backupDetails() and backups().

## Does this PR need a docs update or release note?

- [x] :no_entry: No 

## Type of change

- [x] :broom: Tech Debt/Cleanup

## Issue(s)

* #1970

## Test Plan

- [x] :zap: Unit test
- [x] :green_heart: E2E

### CODE REVIEW

**Review by neha-Gupta1 — APPROVED** [review]

File: src/cli/backup/exchange.go [hunk 1]
```diff
@@ -502,13 +503,14 @@ func runDetailsExchangeCmd(
 		return nil, errs.Fail(err)
 	}
 
-	d, _, err := r.BackupDetails(ctx, backupID)
-	if err != nil {
-		if errors.Is(err, kopia.ErrNotFound) {
+	d, _, errs := r.BackupDetails(ctx, backupID)
+	// TODO: log/track recoverable errors
+	if errs.Err() != nil {
+		if errors.Is(errs.Err(), kopia.ErrNotFound) {
```
Comment (dadams39) [inline src/cli/backup/exchange.go:509]: Do we need to add Clues here?

File: src/cli/backup/exchange_integration_test.go [hunk 2]
```diff
@@ -396,8 +397,9 @@ func (suite *PreparedBackupExchangeIntegrationSuite) TestExchangeDetailsCmd() {
 			bID := suite.backupOps[set]
 
 			// fetch the details from the repo first
-			deets, _, err := suite.repo.BackupDetails(ctx, string(bID))
-			require.NoError(t, err)
+			deets, _, errs := suite.repo.BackupDetails(ctx, string(bID))
+			require.NoError(t, errs.Err())
+			require.Empty(t, errs.Errs())
```
Comment (dadams39) [inline src/cli/backup/exchange_integration_test.go:402]: Should this be `deets` also?

File: src/cli/backup/onedrive.go [hunk 3]
```diff
@@ -394,13 +395,14 @@ func runDetailsOneDriveCmd(
...
-		if errors.Is(err, kopia.ErrNotFound) {
+	d, _, errs := r.BackupDetails(ctx, backupID)
+	// TODO: log/track recoverable errors
+	if errs.Err() != nil {
+		if errors.Is(errs.Err(), kopia.ErrNotFound) {
 			return nil, errs.Fail(errors.Errorf("no backup exists with the id %s", backupID))
 		}
 
-		return nil, errs.Fail(errors.Wrap(err, "Failed to get backup details in the repository"))
+		return nil, errs.Fail(errors.Wrap(errs.Err(), "Failed to get backup details in the repository"))
```
Comment (dadams39) [inline src/cli/backup/onedrive.go:405]: Should the original error be included in L401?

File: src/cli/utils/testdata/opts.go [hunk 4]
```diff
@@ -497,8 +498,11 @@ func (MockBackupGetter) Backup(
...
 }
 
-func (MockBackupGetter) Backups(context.Context, []model.StableID) ([]*backup.Backup, error) {
-	return nil, errors.New("unexpected call to mock")
+func (MockBackupGetter) Backups(
+	context.Context,
+	[]model.StableID,
+) ([]*backup.Backup, *fault.Errors) {
+	return nil, fault.New(false).Fail(errors.New("unexpected call to mock"))
 }
```
Comment (dadams39) [inline src/cli/utils/testdata/opts.go:506]: The naming is interesting because it always returns an error.

**Review by dadams39 — APPROVED** [review]
There are a few nits, but nothing to stop the process.

File: src/cli/utils/testdata/opts.go (hunk 4 above)
Comment (ryanfkeepers) [reply src/cli/utils/testdata/opts.go:506]: It's a bit of a standard pattern for mocks.  The mock has to comply with an interface; but if certain calls on the interface aren't expected, it returns an error to showcase that the mock isn't set up to handle that function.

File: src/cli/backup/exchange.go (hunk 1 above)
Comment (ryanfkeepers) [reply src/cli/backup/exchange.go:509]: Good question, and no.  The general handling is that if were passing a `ctx` into an internal func, then we should assume _that_ func handles packing clues into the error.  There's no harm if we over-pack the data.  But this helps minimize boilerplate.

File: src/cli/backup/exchange_integration_test.go (hunk 2 above)
Comment (ryanfkeepers) [reply src/cli/backup/exchange_integration_test.go:402]: I think deets gets used below, and is therefore checked in that way.

File: src/cli/backup/onedrive.go (hunk 3 above)
Comment (ryanfkeepers) [reply src/cli/backup/onedrive.go:405]: It is the original error

**Pull request merged**